ddocs/erlang_filters/filters/doc_by_place.erl

```diff
@@ -0,0 +1,98 @@
+% Benchmarking code for Erlang REPL nicked and modified to remove module param from here:
+%  https://erlangcentral.org/wiki/index.php/Measuring_Function_Execution_Time
+% Params: FnName, Params, Rounds, eg:
+%  TC(DeepGet, [DeepGet, [<<"userCtx">>, <<"name">>], Req], 100000).
+% TC = fun(TC_F, TC_A, TC_N) when TC_N > 0 -> TC_L = tl([begin {TC_T, _Result} = timer:tc(TC_F, TC_A), TC_T end || _ <- lists:seq(1, TC_N)]), TC_Min = lists:min(TC_L), TC_Max = lists:max(TC_L), TC_Med = lists:nth(round((TC_N - 1) / 2), lists:sort(TC_L)), TC_Avg = round(lists:foldl(fun(TC_X, TC_Sum) -> TC_X + TC_Sum end, 0, TC_L) / (TC_N - 1)), io:format("Range: ~b - ~b mics~nMedian: ~b mics ~nAverage: ~b mics~n", [TC_Min, TC_Max, TC_Med, TC_Avg]), TC_Med end.
+
+% Made "cmd+v futon-ready" with the following disgusting display of black magic
+% cat scratch.erl | sed 's/%.*//' | sed 's/"/\\"/'g | sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/ /g' | pbcopy
+fun ({Doc}, {Req}) ->
+
+  % Some values are wrapped in a tuple for some arbitrary reason
+  %  i.e. {[{...}]} instead of [{...}]
+  % Instead of working this out we're just guarding against it
+  % TODO work out all the rules for this so we can optimise this function
+  %      away for better performance
+  SafeGetValue = fun(Key, PropListMaybe, Default) ->
+    if
+      is_tuple(PropListMaybe) andalso size(PropListMaybe) == 1 ->
+        proplists:get_value(Key, element(1, PropListMaybe), Default);
+      is_list(PropListMaybe) ->
+        proplists:get_value(Key, PropListMaybe, Default);
+      true ->
+        Default
+    end
+  end,
+
+  DeepGet = fun(DeepGet, KeyList, PropList) ->
+    if
```

Comment:
`KeyList` is data that we provide, so I didn't guard against someone using the function incorrectly. I figured that people should use the code correctly, and if they don't it will insta-fail when it tries to pull the head and tail out of it (both those functions fail if you provide them `[]`). It will fail with a badarg into the couchdb console, so it should be reasonably understandable. We could explicitly guard against it, but yeah, I don't actually know how that works in Erlang :P It's also more code, which may or may not affect performance (honestly not sure).